### DISCUSSION TITLE
Multi dispatch for methods

### DISCUSSION THREAD

**Discussion by otchita** [coady/multimethod#40] [Q&A]
Hello,

As I was using this library for multi dispatching for functions, I was wondering if the same functionality can be added to methods as well. It would look some like this:

```python
class Test:

    @multidispatchmethod
    def meth(self, *args, **kwargs):
        ...

    @meth.register(int, float)
    def _(self, x: int, y: float):
        ...

    @meth.register(str, str)
    def _(self, x: str, y: str):
        ...
```

I would be open to work on this eventual feature 😄 .

**Comment by coady** [comment]
Hi. It already supports methods:
```python
@meth.register
def _(self, x: int, y: float):
    ...
```
